crates/hir_ty/src/display.rs

```diff
@@ -313,14 +332,18 @@ impl HirDisplay for ApplicationTy {
                     let ret_display = if f.omit_verbose_types() {
                         ret.display_truncated(f.db, f.max_size)
                     } else {
-                        ret.display(f.db)
+                        if f.display_target.is_test() {
```

Comment:
hmm, these seem like they were already bugs. We should be reusing the whole `HirFormatter`. Anyway, we can look into that separately.